code/modules/assembly/flash.dm

```diff
@@ -311,3 +312,56 @@
 /obj/item/assembly/flash/shield/hit_reaction(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", final_block_chance = 0, damage = 0, attack_type = MELEE_ATTACK)
 	activate()
 	return ..()
+	
+/obj/item/assembly/flash/hypnotic
+	desc = "A modified flash device, programmed to emit a sequence of subliminal flashes that can send a vulnerable target into a hypnotic trance."
+	flashing_overlay = "flash-hypno"
+	light_color = LIGHT_COLOR_PINK
+	cooldown = 20
+	
+/obj/item/assembly/flash/hypnotic/burn_out()
+	return
+	
+/obj/item/assembly/flash/hypnotic/flash_carbon(mob/living/carbon/M, mob/user, power = 15, targeted = TRUE, generic_message = FALSE)
+	if(!istype(M))
+		return
+	if(user)
+		log_combat(user, M, "[targeted? "hypno-flashed(targeted)" : "hypno-flashed(AOE)"]", src)
+	else //caused by emp/remote signal
+		M.log_message("was [targeted? "hypno-flashed(targeted)" : "hypno-flashed(AOE)"]",LOG_ATTACK)
+	if(generic_message && M != user)
+		to_chat(M, "<span class='disarm'>[src] emits a soothing light...</span>")
+	if(targeted)
+		if(M.flash_act(1, 1))
+			var/hypnosis = FALSE
+			if(M.hypnosis_vulnerable())
+				hypnosis = TRUE
+			if(user)
+				visible_message("<span class='disarm'>[user] blinds [M] with the flash!</span>")
+				to_chat(user, "<span class='danger'>You hypno-flash [M]!</span>")
+				if(!hypnosis)
+					to_chat(M, "<span class='notice'>The light makes you feel oddly relaxed...</span>")
+			else if(!hypnosis)
+				to_chat(M, "<span class='notice'>The light makes you feel oddly relaxed...</span>")
+				
+			if(hypnosis) //Placed here so the trance message does not appear before the flash message
+				M.apply_status_effect(/datum/status_effect/trance, 200, TRUE)
+			else
+				M.confused += min(M.confused + 10, 20)
+				M.dizziness += min(M.dizziness + 10, 20)
+				M.drowsyness += min(M.drowsyness + 10, 20)
+				M.apply_status_effect(STATUS_EFFECT_PACIFY, 100)
+				
+		else if(user)
+			visible_message("<span class='disarm'>[user] fails to blind [M] with the flash!</span>")
+			to_chat(user, "<span class='warning'>You fail to hypno-flash [M]!</span>")
+			to_chat(M, "<span class='danger'>[user] fails to blind you with the flash!</span>")

```

Comment:
Is the text from the `visible_message` and second `to_chat` duplicated? If so, add a new proc for them on `/flash`